src/buildercore/core.py

```diff
@@ -279,6 +279,23 @@ def stack_conn(stackname, username=config.DEPLOY_USER, node=None, **kwargs):
 class NoPublicIps(Exception):
     pass
 
+def all_node_params(stackname):
+    data = stack_data(stackname)
+    public_ips = {ec2['InstanceId']: ec2.get('PublicIpAddress') for ec2 in data}
+    nodes = {ec2['InstanceId']: int(tags2dict(ec2['Tags'])['Node']) if 'Node' in tags2dict(ec2['Tags']) else 1 for ec2 in data}
```

Comment:
Not sure about this line. Might need a comment for context or splitting over multiple lines for readability.